Mathlib/FieldTheory/Differential/Liouville.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.Algebra.Field
+import Mathlib.FieldTheory.Differential.Basic
+import Mathlib.FieldTheory.Galois.Basic
+import Mathlib.FieldTheory.IsAlgClosed.AlgebraicClosure
+
+/-!
+# Liouville's theorem
+
+A proof of Liouville's theorem. Follows
+[Rosenlicht, M. Integration in finite terms][Rosenlicht_1972].
+
+## Liouville field extension
+
+This file defines Liouville field extensions, which are field extensions which satisfy
+a slight generalization of Liouville's theorem. Note that this definition doesn't appear in the
+literature, and we introduce it as part of the formalization of Liouville's theorem.
+-/
+
+open Differential algebraMap IntermediateField Finset Polynomial
+
+variable (F : Type*) (K : Type*) [Field F] [Field K] [Differential F] [Differential K]
+variable [Algebra F K] [DifferentialAlgebra F K]
+
+/--
+We say that a field extension `K / F` is Liouville if, whenever an element a ∈ F can be written as
+`a = v + ∑ cᵢ * logDeriv uᵢ` for `v, cᵢ, uᵢ ∈ K` and `cᵢ` constant, it can also be written in that
+way with `v, cᵢ, uᵢ ∈ F`.
+-/
+class IsLiouville : Prop where
```

Comment:
I'm not good for judging that but you make this a class, and the Liouville hypotheses all appear as typeclass arguments. Given the number of instances that you add, is there a risk that the machine will struggle to find the correct path?